server/services/SsotToRobotParsing/retrieveParameters.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @category Server
+ * @module
+ */
+const mongoose = require('mongoose');
+const { ACTIVITY_IDENTIFIER } = require('./robotCodeConstants');
+// eslint-disable-next-line no-unused-vars
+const ssotModels = require('../../models/singleSourceOfTruthModel.js');
+// eslint-disable-next-line no-unused-vars
+const jobsModel = require('../../models/robotJobModel.js');
+
+/**
+ * @description For all activities in the ssot this method will retrieve the associated parameter objects
+ * @param {Object} ssot The ssot for which the parameters should be retrieved
+ * @returns {Array} Array of parameter objects
+ */
+const retrieveParameters = async (ssot) => {
+  const { id } = ssot;
+  const { elements } = ssot;
+  const listOfActivityIds = [];
+
+  elements.forEach((element) => {
+    if (element.type === ACTIVITY_IDENTIFIER) {
+      listOfActivityIds.push(element.id);
+    }
+  });
+
+  const parameterObjects = await mongoose
+    .model('parameter')
+    .find(
+      {
+        robotId: id,
+        activityId: { $in: listOfActivityIds },
+      },
+      {
+        activityId: 1,
+        rpaParameters: 1,
+        outputVariable: 1,
+      }
+    )
+    .exec();
+
+  return parameterObjects;
+};
+
+/**
+ * @description Update Parameter Objects with new parameters
+ * @param {String} parameterObjects The selection of parameter objects this function will have a look at
+ * @param {String} newParameters New parameters in the form {id, value} that the function will use to update the parameter objects
+ */
+const updateParameterObjects = (parameterObjects, newParameters) => {
```

Comment:
Not use, if this couldn't be done in an async method, would again remove array.prototype

Reply:
✅

Reply:
✅